src/expand.cpp

```diff
@@ -740,7 +740,7 @@ static expand_result_t expand_cmdsubst(wcstring input, const operation_context_t
         wcstring sub_res_joined;
         sub_res_joined.reserve(approx_size);
         for (const wcstring &line : sub_res) {
-            sub_res_joined.append(escape_string_for_double_quotes(std::move(line)));
+            sub_res_joined.append(escape_string_for_double_quotes(line));
```

Comment:
Umm this warning was introduced by faf51e0693b73ceae21a1ed34a6ac452b96cace6
I forgot that we do want to move out of `sub_res` here to avoid a copy, so we can't use `const wcstring &` but rather `wcstring &`. If clang-tidy suggested `const wcstring &` here, that's a bug IMO.

Reply:
Fixed. I probably manually did it to be consistent with the surrounding code.

Generally speaking, auto&& should be used in for loops like this. That way you don't worry around bad transformations.

auto&& in for range loops is not the same as a function parameter as mentioned in the google style doc.

Reply:
> bad transformations.

can you give an example of what can go  wrong when using `auto &`?
This might be a good addition to that style guide.

Reply:
auto&& transforms into either auto& or const auto&, with the latter being preferred.

When it encounters an std::move, the compiler knows to evaluate to auto& and clang-tidy will not remove the std::move like it did here.

Note that I'm talking about for range loops. function parameters are a different beast.

As a sidenote: In a different project, I transformed all for range loops to auto&& and got a size reduction. I haven't tried here.

Reply:
I'd keep using `const auto &` over `auto &&` because it prevents us from accidentally modifying the value.
Then it only makes sense to also use `auto &` which is simpler and I haven't found any downsides.

>  I transformed all for range loops to auto&& and got a size reduction. I haven't tried here.

Wow, I'd love to find a small example that shows how this changes the generated code.

Reply:
If you're curious: https://github.com/gerbera/gerbera/commit/66b3c7736f2ffa7e1faa98ec2d62804ecb555683